### PR TITLE
Add recommendations related to symmetric encryption configuration [HZ-4405] [HZ-4406]

### PR DESCRIPTION
**Pull request by kwart**
This PR adds recommendations related to symmetric encryption configuration.

Related JIRAs: [HZ-4405](https://hazelcast.atlassian.net/browse/HZ-4405), [HZ-4406](https://hazelcast.atlassian.net/browse/HZ-4406)

[HZ-4405]: https://hazelcast.atlassian.net/browse/HZ-4405?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[HZ-4406]: https://hazelcast.atlassian.net/browse/HZ-4406?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/modules/security/pages/encryption.adoc [hunk 1]
```diff
@@ -91,6 +91,9 @@ You set the encryption algorithm, the salt, password and the iteration count to
 for generating the secret key. You also need to set the `enabled` attribute to `true`.
 Note that all members should have the same encryption configuration.
 
+NOTE: Use a strong encryption algorithm. Refer to the latest NIST guidelines for additional guidance.
```
Comment (rebekah-lawrence) [inline docs/modules/security/pages/encryption.adoc:94]: Do we want to add a link to the NIST site: https://www.nist.gov/privacy-framework/resource-repository/browse/guidelines-and-tools ?

File: docs/modules/storage/pages/configuring-persistence.adoc [hunk 2]
```diff
@@ -831,6 +831,9 @@ a|'' (empty)
 
 |===
 
+NOTE: Use a strong encryption algorithm. Refer to the latest NIST guidelines for additional guidance.
```
Comment (rebekah-lawrence) [inline docs/modules/storage/pages/configuring-persistence.adoc:834]: Same here, add link?

**Review by rebekah-lawrence — APPROVED** [review]
I have asked whether we need to add links to the NIST site, otherwise LGTM so approving.

**Comment by kwart** [comment]
I would leave it either without the link or used a more generic one - e.g. https://www.nist.gov/cryptography

WDYT, @rebekah-lawrence ?

**Comment by rebekah-lawrence** [comment]
Links to external sites are always difficult - but this site looks like it'll stay about!  I think, to save the reader a search, a link is a good thing, if readers know, they won't notice it, but if they need further information they can easily get there. 

The general link looks good to me.

**Comment by kwart** [comment]
Thanks, Rebekah. I've added the link in a new commit.

**Pull request merged**